### PR TITLE
Rewrite drawing

### PR DESCRIPTION
**Pull request by vpzomtrrfrt**
It's happening.

An IPainter is an object capable of painting circles and text using a PaintStyle.  A PaintStyle holds information about how to paint things, like color, whether it's an outline, etc.

### CODE REVIEW

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    private double scale;
+
+    public GraphicsHandler(GameHandler gameHandler) {
+        this.gameHandler = gameHandler;
+    }
+
+    public void paint(IPainter g) {
+        checkScaling(g);
+        java.util.List<Node> nodes = gameHandler.getNodes();
+        java.util.List<UnitGroup> groups = gameHandler.getUnitGroups();
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l26)

File: test/src/fictionalpancake/turbospork/ColorTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        int a = rand.nextInt(256);
+        Color c = new Color(r, g, b, a);
+        Assert.assertEquals(c.getRed(), r);
+        Assert.assertEquals(c.getGreen(), g);
+        Assert.assertEquals(c.getBlue(), b);
+        Assert.assertEquals(c.getAlpha(), a);
+    }
+
+    @Test
+    public void testNumber() {
```
Comment (vpzomtrrfrt) [inline test/src/fictionalpancake/turbospork/ColorTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964921/issues/source?bid=3248828&fileBranchId=3310821#l37)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    private int yOffset;
+    private double scale;
+
+    public GraphicsHandler(GameHandler gameHandler) {
+        this.gameHandler = gameHandler;
+    }
+
+    public void paint(IPainter g) {
+        checkScaling(g);
+        java.util.List<Node> nodes = gameHandler.getNodes();
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l25)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                lastSelected = node;
+            }
+        }
+        else if(selectedNode != null && selectedNode == node) {
+            selectedNode = null;
+        }
+    }
+
+    public void attack(Node node) {
+        if (selectedNode != null && node != null && node != selectedNode) {
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l162)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+    public void select(Node node) {
+        if (selectedNode == null && node != null && gameHandler.isInProgress()) {
+            int pos = gameHandler.getPosition();
+            if ((pos != -1 && pos == node.getOwner()) || node.getUnits(pos) > 0) {
+                selectedNode = node;
+                lastSelected = node;
+            }
+        }
+        else if(selectedNode != null && selectedNode == node) {
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l156)

File: src/fictionalpancake/turbospork/gui/GameMainPanel.java [hunk 6]
```diff
@@ -1,182 +1,37 @@
 package fictionalpancake.turbospork.gui;
 
 import fictionalpancake.turbospork.*;
+import fictionalpancake.turbospork.paint.*;
+import fictionalpancake.turbospork.paint.Color;
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/gui/GameMainPanel.java]: Codacy Issue found: [Avoid unused imports such as 'fictionalpancake.turbospork.paint.Color'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964910/issues/source?bid=3248828&fileBranchId=3310821#l5)

**Comment by riftEmber** [comment]
Well I guess the stuff I wrote for yo moms house client side is gone now lol

**Comment by vpzomtrrfrt** [comment]
AAAAAAAAAAAAAAAAAAAAAAAA

File: test/src/fictionalpancake/turbospork/ColorTest.java (hunk 2 above)
Comment (vpzomtrrfrt) [inline test/src/fictionalpancake/turbospork/ColorTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964921/issues/source?bid=3248828&fileBranchId=3310821#l37)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java (hunk 1 above)
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l26)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java (hunk 3 above)
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l25)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java (hunk 5 above)
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l156)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java (hunk 4 above)
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964917/issues/source?bid=3248828&fileBranchId=3310821#l162)

File: src/fictionalpancake/turbospork/gui/GameMainPanel.java (hunk 6 above)
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/gui/GameMainPanel.java]: Codacy Issue found: [Avoid unused imports such as 'fictionalpancake.turbospork.paint.Color'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610964910/issues/source?bid=3248828&fileBranchId=3310821#l5)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+    private double scale;
+
+    public GraphicsHandler(GameHandler gameHandler) {
+        this.gameHandler = gameHandler;
+    }
+
+    public void paint(IPainter g) {
+        checkScaling(g);
+        java.util.List<Node> nodes = gameHandler.getNodes();
+        java.util.List<UnitGroup> groups = gameHandler.getUnitGroups();
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610985400/issues/source?bid=3248828&fileBranchId=3310821#l26)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+    private int yOffset;
+    private double scale;
+
+    public GraphicsHandler(GameHandler gameHandler) {
+        this.gameHandler = gameHandler;
+    }
+
+    public void paint(IPainter g) {
+        checkScaling(g);
+        java.util.List<Node> nodes = gameHandler.getNodes();
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2610985400/issues/source?bid=3248828&fileBranchId=3310821#l25)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+        java.util.List<UnitGroup> groups = gameHandler.getUnitGroups();
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2611259179/issues/source?bid=3248828&fileBranchId=3310821#l26)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+    private int yOffset;
+    private double scale;
+
+    public GraphicsHandler(GameHandler gameHandler) {
+        this.gameHandler = gameHandler;
+    }
+
+    public void paint(IPainter g) {
+        checkScaling(g);
+        java.util.List<Node> nodes = gameHandler.getNodes();
```
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2611259179/issues/source?bid=3248828&fileBranchId=3310821#l25)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java (hunk 9 above)
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java:26]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2628516103/issues/source?bid=3248828&fileBranchId=3310821#l26)

File: src/fictionalpancake/turbospork/paint/GraphicsHandler.java (hunk 10 above)
Comment (vpzomtrrfrt) [inline src/fictionalpancake/turbospork/paint/GraphicsHandler.java:25]: Codacy Issue found: [Unnecessary use of fully qualified name 'java.util.List' due to existing import 'java.util.List'](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork/file/2628516103/issues/source?bid=3248828&fileBranchId=3310821#l25)

**Pull request merged**